packages/@uppy/companion/src/server/provider/webdav/index.js

```diff
@@ -115,8 +115,17 @@ export default class WebdavProvider extends Provider {
           // ignore invalid date from server
         }
 
+        // Determine icon based on type and MIME type
+        let icon = 'file'
+        if (isFolder) {
+          icon = 'folder'
+        } else if (item.mime?.startsWith('video/')) {
+          icon = 'video'
+        }
```

Comment:
do we also have an `image` icon type? if so, should we add it too? other than that lgtm